26th July 2014 3:09 AM IST

Today I worked to make sure that `solve_univariate` returns a `RootOf`
solution if it cannot solve a given polynomial. First I tried to replace all
roots instances with `Poly(f, x).all_roots()` because `all_roots()` is not able
to solve all the equations `roots` can solve. For example it failed on
`Poly(a*x**2 + b*x + c, x, domain='EX').all_roots()` so I decided to call it
only when the roots solver failed.

There was another issue, one of the tests I imported was wrongly imported. This
time I figured it out early but it can dangerous.

Then git commits should not contain anything unrelated to the what the commit
message says it does. Not doing so is becomes problematic in rebase and also it
becomes hard to track down the changes
